lib/Dialect/TritonGPU/Transforms/Pipeliner/MatmulLoopPipeline.cpp

```diff
@@ -448,9 +448,10 @@ assignMemoryLayouts(llvm::SmallVector<std::tuple<Operation *, int, Operation *>>
         loadInfo.sharedEncoding =
             getSharedEncIfAllUsersAreDotEnc(op->getResult(0), incompatible)
                 .value_or(nullptr);
-        // If we can't agree on a shared encoding skip pipelinig the load.
-        if (incompatible)
-          continue;
+        // Incompatible shared encoding has been handled in
+        // splitLoadsForIncompatible.
+        assert(!incompatible && "incompatible shared encoding");
```

Comment:
Since this is should never happen, we could assert in the `getSharedEncIfAllUsersAreDotEnc` and simplify the code by not having to return it here.